### ISSUE TITLE
WYSIWYG lights using Blender's UI for spill, custom, param lights

### ISSUE THREAD

**Issue by danklaue** [X-Plane/XPlane2Blender#183]
Suggest using Blender's spot lights for this. Those parameters can be exported and translated for X-Plane's lighting usage.  

For instance, 
-"spot shape" (in Blender) can be translated for X-Plane's spill lights, as the cone size and angle of the spill. It could also be used in X-Plane's custom lights for custom lights' W value (Width).
-"Energy" (in Blender) can be translated into X-Plane's spill lights, as size parameter, or for Custom lights for the halo size.
-Directionality is integrated in Blender's spot lights... it would just have to be translated to the OBJ8 lighting direction format.
-An X-Plane override to attach it to brightness ratio dataref.
-Translate the RGB values of the spot light for use in X-Plane.

-Leave current infrastructure as is, for backward compatibility, but use spot lights for "new" export capabilities.

**Comment by bsupnik** [comment]
Ondrej -- we should come up with some specs for this at some point; light integration -is- something that would really bring modeling forward. One of the tricky things is how Blender can know how a light works; my thinking is that we (LR) could provide more useful annotations in lights.txt so that Blender can correctly translate a named or param light, just like Blender (2.49) can use DataRefs.txt to manage animations.

**Comment by jeiting** [comment]
From what I can tell this is untouched? The exporter has basic support for param lights but you need to place one light per param and setting the spill color is not supported. I'll probably come to lights soon on the model I'm working on and maybe I'll pick this up to make my life easier.

**Comment by bsupnik** [comment]
We are planning WYSIWYG light editing including spill as the next major feature once 3-4 is finished.  3-4 will include v11 attribute support and bug fixes for animation and attributes.

**Comment by danklaue** [comment]
Up until recently, one of the parameters that controlled directionality for the "custom lights" in Blender 2.7 was the RGB value of the light in Blender.  E.g. to have a [custom light ](http://developer.x-plane.com/?article=obj8-file-format-specification#LIGHT_CUSTOM_ltxgt_ltygt_ltzgt_ltrgt_ltggt_ltbgt_ltagt_ltsgt_lts1gt_ltt1gt_lts2gt_ltt2gt_ltdatarefgt)with a custom png billboard shine in a particular direction, let's say "forward", one would have to see the exported .obj have something like this in it:

`LIGHT_CUSTOM	-0.00000031 0.33022034 -0.66518688 0 0 1 1 2 0 0.5 0.5 1 sim/graphics/animation/lights/airplane_navigation_light_dir`

where the "0 0 1" after the positional coordinates, according to the documentation represent RGB values (tinting of the halo texture) but actually in practice represent a directionality of the light... in this case, a light shining straight back, with not too much "focus" to the light.  

If one would want this same halo to be visible from the front, one would have to change the "0 0 1" to "0 0 -1", or, if you want the cone where that halo is visible to be tighter, you'd increase the value of the last number in the negative direction , like "0 0 -7".  

This used to be doable via Blender's RGB control sliders.  But a recent version of Blender discontinued support to enter negative values into the RGB slots, so now the exported .obj files have to be hacked, in order to affect directionality of custom light billboards.  

Now, I'm sure this'll be addressed once lights are re-vamped. Just wanted to make Ben and Ted aware of this.

**Comment by tngreene** [comment]
> Up until recently, one of the parameters that controlled directionality for the "custom lights" in Blender 2.7 was the RGB value of the light in Blender...

For reference, this was resolved in #316

**Comment by tngreene** [comment]
This was brought up again in #312. The ideas brought up so far

OBJ Light Concept | Blender UI Mapping
------------------------|-----------------------
X, Y, Z                    | Light Location
DX, DY, DZ             | Light Spot
RGB                       | Light Color Picker
A                            | Since A is just a dimmer, not real transparency, people can just use the color picker and we always set this to 1
Size                       | Light Energy. It is universally located

Some common elements from lights.txt

PARAM | Blender UI Mapping
----------|-------------------------
BRIGHT(see helipad_flood_sp) | Light Size again?
THROW (see heli_3_col_beacon)|?
WIDTH (see helipad_flood_sp) |?
W (aka Width) | See above
FOCUS |?

I'm nervous about intelligently filling in the param def too much because I don't want to frazzle users. 

- "Why isn't helipad_flood_sp working?! Oh, that uses Lamp's Size? Fine"
- "Lamp Energy for Width? That makes no sense to me! Energy was used for something else in another light!"
- "FOCUS means Width means Lamp energy?!"
- "Omni must be a Point Light With Sphere checked? But, I don't want to use a Point Light here!"
"INDEX is where? A new XPlane2Blender prop?"

I don't want to scatter these things around and make people look up what options are relevant when in the manual. Plus, since the names of lights.txt's parameters aren't really standard, I don't want to have to keep making new mappings every new light.

I also don't want a billion new props like with manipulator having a prop for ever single v1,v2,v1_min,v1_max,v2_min,v2_max etc PARAM name that can't have a good Blender mapping. Plus, how would you know what to show in the UI without tieing it in tightly with lights.txt itself.

Either some guarantees are made about names in lights.txt and some REALLY good choices are made for a subset of common light parts, or we stick with just the basics - XYZ from location, RGB from a color picker, A is autoset to 1, DX, DY, DZ from the rotation.

Also, I think there should be a manual override for this.

**Comment by tngreene** [comment]
There is another issue with WYSIWYG - Blender doesn't render things the same as X-Plane. We could be making a promise we can't keep.

**Comment by tngreene** [comment]
There is also Blender 2.8 to think about. Looking casually at it now, the some of the labels and options are different between the two. I vote to move this feature to after the transition has been made.

**Comment by arb65912** [comment]
"I vote to move this feature to after the transition has been made." 
I think since all the hype in now on Blender 2.8 , there is a good chance it will became a standard for many so personally I would like to see transition first.

**Comment by tngreene** [comment]
Merged! 0fb061421ab269fc8

Prepare ye masses, for this awesome feature will soon be upon you in beta.2!
